### PR TITLE
Feat : Add deployment to Firebase via Github Actions

### PR DESCRIPTION
**Pull request by ofcljaved**
# Pull Request Description
This pull request add a feature to deploy backend to firebase via Github Actions .

### Add github workflows
- YAML file check setup the node because backend folder is using npm
- Install the dependencies as turbo doesn't install it for function folder inside backend
- Build the code as turbo build doesn't build it, it only builds web folder
- Deploy to firebase by using github secret of FIREBASE_TOKEN

### Github Marketplace
- Github marketplace also provide a way to upload firebase cloud function
- [Deploy to Firebase Cloud Functions](https://github.com/marketplace/actions/deploy-to-firebase-cloud-functions)

### CODE REVIEW

File: packages/ui/src/Appbar.tsx [hunk 1]
```diff
@@ -40,6 +40,8 @@ export const Appbar = ({ user }: { user: User | null }) => {
                     // Sign-out successful.
                   },
                   function (_error) {
+                    console.log(_error);
```
Comment (hkirat) [inline packages/ui/src/Appbar.tsx:43]: remove

File: packages/ui/src/BlogAppbar.tsx [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
   }, [track, problem]);
 
-  let totalPages = Array.from({ length: track.problems.length }, (_, i) => i + 1);
+  // let totalPages = Array.from({ length: track.problems.length }, (_, i) => i + 1);
+  //not using this so I commented it to avoid eslint-warnings
 
-  function setTheme(arg0: string) {
-    throw new Error("Function not implemented.");
-  }
+  //not using this so I commented it to avoid eslint-warnings
```
Comment (hkirat) [inline packages/ui/src/BlogAppbar.tsx:17]: remove
no need o have these comments

File: packages/ui/src/ErrorBoundary.tsx [hunk 3]
```diff
@@ -7,6 +7,7 @@ export class ErrorBoundary extends Component {
   }
 
   static getDerivedStateFromError(error: any) {
+    console.log(error);
```
Comment (hkirat) [inline packages/ui/src/ErrorBoundary.tsx:10]: remove

**Review by hkirat — CHANGES_REQUESTED** [review]
please keep the PR mimimal
in this case, please make sure it only has ci/cd changes

File: packages/ui/src/Appbar.tsx (hunk 1 above)
Comment (ofcljaved) [reply packages/ui/src/Appbar.tsx]: removed Unnecessary changed

File: packages/ui/src/BlogAppbar.tsx (hunk 2 above)
Comment (ofcljaved) [reply packages/ui/src/BlogAppbar.tsx]: removed Unnecessary changed

File: packages/ui/src/ErrorBoundary.tsx (hunk 3 above)
Comment (ofcljaved) [reply packages/ui/src/ErrorBoundary.tsx]: removed Unnecessary changed

**Comment by ofcljaved** [comment]
@hkirat  I have tried to revert all those changes and keep it to minimum

File: .github/workflows/firebase.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      run: npm install -g firebase-tools
+    
+    - name: Build Backend
+      working-directory: apps/backend/functions
+      run: npm run build  
+    
+    - name: Deploy to Firebase Functions
+      working-directory: apps/backend  
+      run: |
+        firebase login:ci --no-localhost --token "$FIREBASE_TOKEN"  # Login to Firebase using a CI token
```
Comment (hkirat) [inline .github/workflows/firebase.yml:36]: Where should I add FIREBASE_TOKEN  in the repo??

**Review by hkirat — CHANGES_REQUESTED** [review]

File: .github/workflows/firebase.yml (hunk 4 above)
Comment (ofcljaved) [reply .github/workflows/firebase.yml:36]: add in the repository secret
@hkirat

**Comment by hkirat** [comment]
Not using firebase anymore

**Pull request closed without merging**